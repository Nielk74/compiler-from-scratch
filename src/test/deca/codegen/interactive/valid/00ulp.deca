// Description:
//   test ULP
//
// Resultats:
//    00ulp.expected
//
// Historique:
//    cree le 16/06/2022

#include "Math.decah"
{
    Math m = new Math();
    float x;
    int a = 0;
    while(a < 34){
        a = a + 1;
        x = readFloat();
        x = m.ulp(x);
        printlnx(x);
    }
}