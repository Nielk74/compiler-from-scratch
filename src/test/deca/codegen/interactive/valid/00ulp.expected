0x1p-149
0x1p-23
0x1p-22
0x1p-22
0x1p-1
0x1p+0
0x1p+0
0x1p+1
0x1p-24
0x1p-25
0x1p-26
0x1p-26
0x1p-27
0x1p+8
0x1p+104
0x1p-149
0x1p-149
0x1p-149
0x1p-23
0x1p-22
0x1p-22
0x1p-1
0x1p+0
0x1p+0
0x1p+1
0x1p-24
0x1p-25
0x1p-26
0x1p-26
0x1p-27
0x1p+8
0x1p+104
0x1p-149
0x1p-149
